src/gennylib/src/quiesce.cpp

```diff
@@ -94,6 +94,65 @@ bool waitOplog(v1::Topology& topology) {
     return waitVisitor.success();
 }
 
+bool CheckForDroppedCollections(v1::Topology& topology, std::string dbName, SleepContext sleepContext) {
+    class CheckForDroppedCollectionsVisitor : public v1::TopologyVisitor {
+    public:
+        CheckForDroppedCollectionsVisitor(std::string dbName, SleepContext sleepContext)
+            : _dbName{dbName},
+              _sleepContext{sleepContext} {}
+
+        void onBeforeReplSet(const v1::ReplSetDescription& desc) override {
+            mongocxx::client client(mongocxx::uri(desc.primaryUri));
+            _successAcc = checkCollectionsTestDB(client);
+        }
+
+        bool success() {
+            return _successAcc;
+        }
+
+    private:
+        bool checkCollections(mongocxx::database db) {
+            using bsoncxx::builder::basic::kvp;
+            using bsoncxx::builder::basic::make_document;
+
+            auto res = db.run_command(make_document(kvp("serverStatus", 1)));
+            if (!res.view()["storageEngine"] || !res.view()["storageEngine"]["dropPendingIdents"]) {
+                return false;
+            }
+            auto idents = res.view()["storageEngine"]["dropPendingIdents"];
+
+            return idents.get_int64() != 0;
+        }
+
+        bool checkCollectionsTestDB(mongocxx::client& client) {
+            auto db = client.database(_dbName);
+            int retries = 0;
+            while (checkCollections(db) && retries < DROPPED_COLLECTION_RETRIES) {
+                BOOST_LOG_TRIVIAL(debug)
+                    << "Sleeping 5 second while waiting for collection to finish dropping.";
+                retries++;
+                _sleepContext.sleep_for(std::chrono::seconds(5));
```

Comment:
Changed to sleep for 5 seconds instead of 1 as it still works (with the 20 retries) and gives a 100 second limit, instead of 20s.